### PR TITLE
[Feature][shade][Jackson] Add seatunnel-jackson module

### LINKED ISSUE #3936: [Feature][shade][jackson] Add jackson module
### Search before asking

- [X] I had searched in the [feature](https://github.com/apache/incubator-seatunnel/issues?q=is%3Aissue+label%3A%22Feature%22) and found no similar feature requirement.

### Description

As we know, we will add check config logic in connectors, all connectors' check need `seatunnel-common` module and use the ability of jackson. Unfortunately, the version of jackson between seatunnel-common and spark is conflict.

image

I try to add shade config in `seatunnel-common`, but `seatunnel-connector-v2/connector-common` dependency it, compile will failed as the following shown:

image

So use the shade version of jackson in `seatunnel-common` is a best choice.

### Usage Scenario

_No response_

### Related issues

#3800 

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by TyrantLucifer**
## Purpose of this pull request

close #3936 

## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/incubator-seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/incubator-seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/incubator-seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [ ] Update the [`release-note`](https://github.com/apache/incubator-seatunnel/blob/dev/release-note.md).

### CODE REVIEW

**Review by EricJoy2048 — DISMISSED** [review]

File: seatunnel-api/src/main/java/org/apache/seatunnel/api/configuration/util/OptionRule.java [hunk 1]
```diff
@@ -195,8 +195,9 @@ public <T> Builder conditional(@NonNull Option<T> conditionalOption, @NonNull Li
 
         public <T> Builder conditional(@NonNull Option<T> conditionalOption, @NonNull T expectValue, @NonNull Option<?>... requiredOptions) {
             for (Option<?> o : requiredOptions) {
-                verifyDuplicate(o, "ConditionalOption");
```
Comment (ashulin) [inline seatunnel-api/src/main/java/org/apache/seatunnel/api/configuration/util/OptionRule.java]: IMO, submit a new PR to deal with this issue.

Comment (TyrantLucifer) [reply seatunnel-api/src/main/java/org/apache/seatunnel/api/configuration/util/OptionRule.java]: Yup, you are right, I had submitted another pull request #3962 to solve this problem. Temporary I rebase that onto this branch to test CI process.

**Comment by TyrantLucifer** [comment]
@EricJoy2048 @ashulin PTAL.

**Review by Hisoka-X — APPROVED** [review]

**Comment by CalvinKirs** [comment]
should we put all shade jars in this? github.com/apache/incubator-seatunnel-shade @EricJoy2048

**Comment by TyrantLucifer** [comment]
> should we put all shade jars in this? github.com/apache/incubator-seatunnel-shade @EricJoy2048

Yup, all shade dependencies shoule be moved to this repository. But before the shade repository released, main repository can not add shade dependency in pom file. In oreder to not block the progress of support check logic, so I temporary add it in main repository.

**Review by CalvinKirs — APPROVED** [review]

**Comment by CalvinKirs** [comment]
> should we put all shade jars in this? github.com/apache/incubator-seatunnel-shade @EricJoy2048↳

waiting for the first release.

File: tools/dependencies/known-dependencies.txt [hunk 2]
```diff
@@ -50,6 +50,7 @@ protostuff-runtime-1.8.0.jar
 scala-library-2.11.12.jar
 seatunnel-config-base-2.1.1.jar
 seatunnel-config-shade-2.1.1.jar
+seatunnel-jackson-2.3.1-SNAPSHOT-optional.jar
```
Comment (CalvinKirs) [inline tools/dependencies/known-dependencies.txt:53]: we shouldn't add this line

Comment (TyrantLucifer) [reply tools/dependencies/known-dependencies.txt:53]: If we didn't add this，CI will not be passed. Do you have any good advice？

Comment (CalvinKirs) [reply tools/dependencies/known-dependencies.txt:53]: i will take a look later:)

**Pull request merged**